module-4/backend/week_one.md

```diff
@@ -5,17 +5,60 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. What's the most useful thing you learned from completing the intermission week work?
+
+  I really enjoyed learning about non-relational databases with Neo4j and (FINALLY) implementing the Interactive Broker API. It was also very useful to get comfortable with writing python scripts without a web framework. I also took part of a datacamp course, which gave me a good understanding of how to create hacker data and interpret results.
+
 2. What are some tools to help debug JavaScript code?
+
+  - Developer tools (with many different functions)
+  - console.log(...)
+  - debugger
+
 3. What are some tools you need in order to unit test your JavaScript?
+
+  Mocha and chai, as well as decent enough compartmentalization of JavaScript code.
+
 4. What is the syntax for invoking a function?
+
+  invokeThisFunction() -> invokes
+
 5. What is CORS?
+
+  CORS allows HTTP requests to gain access to assets from a different server than the one currently in use. 
+    e.g. "POST" requests to an external API
+
 6. How do we enable CORS?
+
+  - installing browser extension
+  - adding development hash in webpack config file
+  - adding "Allow-Access-Origin": "..." to headers
+
 7. What's `this` in JavaScript?
+
+  `this` refers to the object that a function is called upon, when being referenced within the function declaration block.
+
 8. What is Webpack and why is it useful?
+
+  webpack is a mini web framework that compiles javascript modules to display content in the browser.
+
 9. When/why do you want to use event delegation?
+
+  Event delegation is useful when one wants to add an event listener that targets certain different child nodes of a common parent node. The event can be set to the parent node in question and then filter out events by using event.target and event.currentTarget.
```

Comment:
Yup its also helpful for elements that may or may not currently exist on the page.